windows/CodePush.Net46/UpdateManager.cs

```diff
@@ -0,0 +1,326 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System.IO;
+using System.IO.Compression;
+using System.Threading.Tasks;
+using System.Runtime.CompilerServices;
+using System;
+#if WINDOWS_UWP
+using Windows.Web.Http;
+#else
+using CodePush.Net46.Adapters.Http;
+using PCLStorage;
+#endif
+
+[assembly: InternalsVisibleTo("CodePush.Net46.UnitTest")]
+
+namespace CodePush.ReactNative
+{
+    internal class UpdateManager
+    {
+        #region Internal methods
+
+        internal async Task ClearUpdatesAsync()
+        {
+            await (await GetCodePushFolderAsync().ConfigureAwait(false)).DeleteAsync().ConfigureAwait(false);
+        }
+
+        internal async Task DownloadPackageAsync(JObject updatePackage, string expectedBundleFileName, Progress<HttpProgress> downloadProgress)
+        {
+            // Using its hash, get the folder where the new update will be saved 
+            var codePushFolder = await GetCodePushFolderAsync().ConfigureAwait(false);
+            var newUpdateHash = (string)updatePackage[CodePushConstants.PackageHashKey];
+            var newUpdateFolder = await GetPackageFolderAsync(newUpdateHash, false).ConfigureAwait(false);
+            if (newUpdateFolder != null)
+            {
+                // This removes any stale data in newPackageFolderPath that could have been left
```

Comment:
Maybe rename the variable to something like `existingUpdateFolder` to make it a bit more readable (then declare a new var below where you actually get the new reference to the folder)

---
In reply to: [99630777](https://github.com/Microsoft/react-native-code-push/pull/684#discussion_r99630777) [](ancestors = 99630777)